protocol/src/main/java/org/geysermc/mcprotocollib/protocol/packet/status/clientbound/ClientboundStatusResponsePacket.java

```diff
@@ -31,13 +31,25 @@ public class ClientboundStatusResponsePacket implements MinecraftPacket {
     // vanilla behavior falls back to false if the field was not sent
     private static final boolean ENFORCES_SECURE_CHAT_DEFAULT = false;
 
-    private final @NonNull ServerStatusInfo info;
+    private final @NonNull JsonObject data;
+
+    public ClientboundStatusResponsePacket(@NonNull ServerStatusInfo info) {
+        this(toJson(info));
+    }
 
     public ClientboundStatusResponsePacket(ByteBuf in, MinecraftCodecHelper helper) {
-        JsonObject obj = new Gson().fromJson(helper.readString(in), JsonObject.class);
-        JsonElement desc = obj.get("description");
+        data = new Gson().fromJson(helper.readString(in), JsonObject.class);
```

Comment:
This can probably be `new JsonParser().parse` - it may say that's deprecated but I know Geyser would much appreciate using the deprecated version instead of the modern version not available on older Gson versions.

Reply:
Isn't Gson.fromJson available in old versions?

Reply:
@Camotoy ?

Reply:
Yes, but JsonParser (available since 1.3) is probably a *tad* bit cleaner and Gson doesn't have to look up the type for JsonObject like with fromJson. I see the old code used fromJson as well, so it doesn't really matter.

Reply:
I see, but this ensures upwards support. Newer Gson versions may remove that deprecated method and that's why I wanna avoid using it.

Reply:
It's been deprecated for five years and there's little reason for Gson to remove it. In a couple years when no project will be using an outdated Gson, we can switch to the static version and it won't be an issue.

Reply:
Tried adding it now, but I don't think it'll work without doing workarounds to use Jsonparser.

Reply:
The main issue is that JsonParser#parse actually generates JsonElements, not JsonObjects. So I'd have to make casts. I think Gson can provide nicer error messages than a cast error if we use fromJson.